### ISSUE TITLE
False positive with `ConstructorLeaksThis`

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#655]
Consider this class:
```java
public final class Test {
    public Test() {
        this(0);
    }

    public Test(int i) {
        /* Field assignment irrelevant. */
    }
}
```

Running with `-Xep:ConstructorLeaksThis:ERROR` yields:
```
Test.java:3: error: [ConstructorLeaksThis] Constructors should not pass the 'this' reference out in method invocations, since the object may not be fully constructed.
        this(0);
        ^
    (see http://errorprone.info/bugpattern/ConstructorLeaksThis)
1 error
```

**Comment by krisztiankocsis** [comment]
Have the same, SuppressWarnings also doesn't work, still prints the message in both ERROR and WARN cases.

public final class ConfigurationEntry
{
  @Nonnull
  private final Integer id;
  @Nonnull
  private final Instant lastChanged;
  @Nonnull
  private final Type type;
  @Nullable
  private final String value;

  @SuppressWarnings("ConstructorLeaksThis")
  public ConfigurationEntry(@Nonnull final Integer id, @Nonnull final Type type, @Nullable final String value)
  {
    this(id, type, value, Instant.now());
  }

  ConfigurationEntry(
    @Nonnull final Integer id, @Nonnull final Type type,
    @Nullable final String value, @Nonnull final Instant lastChanged
  )
  {
    this.id = requireNonNull(id, "ID is required");
    this.lastChanged = requireNonNull(lastChanged, "Last changed is required");
    this.type = requireNonNull(type, "Type is required");
    this.value = value;
  }
}

**Comment by krisztiankocsis** [comment]
Update: Warning suppression works if it is put to the enclosing class, not the the affected constructor.

**Comment by Stephan202** [comment]
@krisztiankocsis I filed #735 for that, as it's a separate issue.

**Comment by Stephan202** [comment]
(I just checked: the issue described in this ticket is still valid in version 2.1.1.)

**Comment by vorburger** [comment]
Just hit this as well .. perhaps someone could merge https://github.com/google/error-prone/pull/789 some time, so that so this goes into the next release? Without this, `ConstructorLeaksThis` isn't really usable on our code base.

**Comment by sebright** [comment]
I think this issue was fixed in c2b37f4f5bc86d6736812ff4cc7f099615f561aa.

**Comment by Stephan202** [comment]
@sebright right you are!
